src/esri-loader.ts

```diff
@@ -11,14 +11,60 @@
   limitations under the License.
 */
 
+const DEFAULT_URL = 'https://js.arcgis.com/4.5/';
+
 // get the script injected by this library
 function getScript() {
-  return document.querySelector('script[data-esri-loader]');
+  return document.querySelector('script[data-esri-loader]') as HTMLScriptElement;
+}
+
+// TODO: at next breaking change replace the public isLoaded() API with this
+function _isLoaded() {
+  // TODO: instead of checking that require is defined, should this check if it is a function?
+  return typeof window['require'] !== 'undefined';
```

Comment:
Again something I just debugged recently in esri-promise, if someone has already included another loader or another package that claims `window.require` then this will return true even though the require is most likely not dojo's. Now in esri-promise I'm just making sure that the `on` method from dojo's require is present, otherwise assume `window.require` is not dojo's `require` and that the JS API still needs to be loaded..

It's not great, since forcing users to have dojo's require instead of whatever one they've already loaded may create problems in their app. On the other hand though, allowing them to use a non-dojo require will break things in the JS API, so it's a bit of a pickle.

I'm currently looking into ways of keeping dojo and the JS API out of the global namespace completely without having to sacrifice any functionality

Reply:
Thanks for the suggestion! I've added this in 
4ebb65d

> I'm currently looking into ways of keeping dojo and the JS API out of the global namespace completely without having to sacrifice any functionality

💯 ☝️ that would be amazing and solve most of my web dev problems.